src/background/services/wallet/handlers/importLedger.ts

```diff
@@ -117,7 +117,14 @@ export class ImportLedgerHandler implements HandlerType {
       });
     }
 
-    await this.#addAccounts(id, numberOfAccountsToCreate || 3);
+    const defaultAccountsToCreate = 3;
+    const accountsToCreate = !pubKeys
+      ? numberOfAccountsToCreate || defaultAccountsToCreate
+      : numberOfAccountsToCreate || defaultAccountsToCreate <= pubKeys.length
+        ? numberOfAccountsToCreate || defaultAccountsToCreate
+        : pubKeys.length;
```

Comment:
So, this repeats `numberOfAccountsToCreate || defaultAccountsToCreate` three times, which makes it a bit hard to read. Can we simplify?

If we have `pubKeys` (Ledger Live), we want to create `<numberOfAccountsToCreate>` accounts, but no more than `pubKeys.length` and never more than 3, right?

And if we don't have `pubKeys` (BIP44), we want to create `<numberOfAccountsToCreate> (if known and above zero), `<defaultAccountsToCreate>` otherwise, but still never more than 3?

So how about this?

```suggestion
    const accountsToCreate = Math.min(
      3, // never create more than 3 accounts
      pubKeys // if we use Ledger Live...
        ? Math.min(pubKeys.length, accountsToBeCreated) // ...go with accountsToBeCreated, but never exceed pubKeys.length...
        : accountsToBeCreated // ... and if we're on BIP44, go with accountsToBeCreated
    );
```